src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -0,0 +1,99 @@
+import logging
+from datetime import datetime
+from typing import Dict, List, Tuple, TypedDict
+
+from django.db import transaction
+
+from sentry.issues.escalating import GroupsCountResponse, query_groups_past_counts
+from sentry.issues.escalating_issues_alg import generate_issue_forecast
+from sentry.models import Group, GroupStatus
+from sentry.models.group import GroupSubStatus
+from sentry.models.groupforecast import GroupForecast
+from sentry.tasks.base import instrumented_task
+
+BATCH_SIZE = 1000
+
+
+class GroupCount(TypedDict):
+    intervals: List[str]
+    data: List[int]
+
+
+ParsedGroupsCount = Dict[int, GroupCount]
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.tasks.weekly_escalating_forecast.run_escalating_forecast",
+    queue="weekly_escalating_forecast",
+    max_retries=0,  # TODO: Increase this when the task is changed to run weekly
+)  # type: ignore
+def run_escalating_forecast() -> None:
+    # TODO: Do not limit to project id = 1 and limit 10 once these topics are clarified
+    until_escalating_groups = list(
+        Group.objects.filter(
+            status=GroupStatus.IGNORED,
+            substatus=GroupSubStatus.ARCHIVED_UNTIL_ESCALATING,
+            project__id=1,
+        )[:10]
+    )
+    if not until_escalating_groups:
+        return
+
+    response = query_groups_past_counts(until_escalating_groups)
+    group_counts = parse_groups_past_counts(response)
+    group_forecast_list = get_forecast_per_group(until_escalating_groups, group_counts)
+
+    with transaction.atomic():
+        # Delete old forecasts
+        GroupForecast.objects.filter(group__in=until_escalating_groups).delete()
+
+        # Bulk create new forecasts
+        GroupForecast.objects.bulk_create(
+            [
+                GroupForecast(group=group, forecast=forecast)
+                for group, forecast in group_forecast_list
+            ],
+            batch_size=BATCH_SIZE,
+        )
```

Comment:
I think that we're only using the `group_id` and maybe the `project_id` here? Once we're doing this at a higher scale, it'll be helpful to just fetch the data we need instead of the entire model. Doesn't need to be in this pr if you just want to get it merged, but leave a todo

Reply:
I believe we might still need the whole `group` in order to create `GroupForecast`, as it uses `group` as the key

Reply:
Thank you, I've added this as a TODO